### PR TITLE
Assure that ASC names are unique when using deployer.

### PR DESCRIPTION
**Pull request by ratik21**

### CODE REVIEW

**Review by robert-zaremba — COMMENTED** [review]
@sczembor could you also review?

File: packages/algob/test/internal/deployer.ts [hunk 1]
```diff
@@ -523,6 +523,42 @@ describe("DeployerDeployMode", () => {
...
+        txID: "",
+        confirmedRound: 0,
+        appID: -1,
+        timestamp: 1,
+        deleted: false,
+        approvalFile: "approval-file.py",
+        clearFile: "clear-file.py"
+      })
+      .putMetadata(networkName, "k", "v");
+    const deployerCfg = new DeployerConfig(env, new AlgoOperatorDryRunImpl());
```
Comment (sczembor) [inline packages/algob/test/internal/deployer.ts:547]:
```suggestion
    const deployerConfig = new DeployerConfig(env, new AlgoOperatorDryRunImpl());
```

File: packages/algob/test/internal/deployer.ts [hunk 2]
```diff
@@ -523,6 +523,42 @@ describe("DeployerDeployMode", () => {
...
+    deployerCfg.cpData = cpData;
```
Comment (sczembor) [inline packages/algob/test/internal/deployer.ts:548]:
```suggestion
    deployerConfig.cpData = cpData;
```

File: packages/algob/test/internal/deployer.ts [hunk 3]
```diff
@@ -523,6 +523,42 @@ describe("DeployerDeployMode", () => {
...
+    const deployer = new DeployerDeployMode(deployerCfg);
```
Comment (sczembor) [inline packages/algob/test/internal/deployer.ts:549]:
```suggestion
    const deployer = new DeployerDeployMode(deployerConfig);
```

**Review by sczembor — APPROVED** [review]

File: packages/algob/src/internal/deployer.ts [hunk 4]
```diff
@@ -533,6 +533,32 @@ export class DeployerDeployMode extends DeployerBasicMode implements Deployer {
     }
   }
 
+  /**
+   * Asserts if lsig is not already present in checkpoint
+   * @param lsigName lsig name
+   */
+  assertNoLsig (lsigName: string): void {
```
[the remark this answers is not in the document]
Comment (ratik21) [reply packages/algob/src/internal/deployer.ts:540]: From the user perspective errors will be different, so created separate functions for app, lsig.

**Pull request merged**